chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -249,23 +249,28 @@ out:
 				}
 				b.spendNotifications[op][msg.spendID] = msg
 				b.chainConn.NotifySpent([]*wire.OutPoint{&op})
-			case *confirmationsNotification:
-				chainntnfs.Log.Infof("New confirmations "+
+			case *confirmationNotification:
+				chainntnfs.Log.Infof("New confirmation "+
 					"subscription: txid=%v, numconfs=%v",
 					msg.TxID, msg.NumConfirmations)
 
+				b.heightMtx.RLock()
+				currentHeight := uint32(b.bestHeight)
+				b.heightMtx.RUnlock()
+
 				// Lookup whether the transaction is already included in the
 				// active chain.
-				txConf, err := b.historicalConfDetails(msg.TxID)
+				txConf, err := b.historicalConfDetails(
+					msg.TxID, msg.heightHint, currentHeight,
+				)
 				if err != nil {
 					chainntnfs.Log.Error(err)
 				}
-				b.heightMtx.RLock()
+
```

Comment:
Why is the mutex around access to the `txConfNotifier` no longer needed?

Reply:
The mutex is only being used to guard the `ChainNotifier`'s `bestHeight`. Calling `txConfNotifier.Register` without it should be fine since it doesn't have an effect on the `bestHeight`. I'm guessing it was needed at some point and it didn't get removed. I could be wrong though.